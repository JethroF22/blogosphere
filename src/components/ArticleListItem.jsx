import React from "react";
import moment from "moment";

const ArticleListItem = props => (
  <div className="uk-card uk-card-default uk-card-hover uk-margin-medium-bottom">
    <div className="uk-card-header">
      <h3 className="article-list__title">{props.post.title}</h3>
      <p className="article-list__subtitle">
        Published on {moment(props.post.createdAt).format("MMMM Do YYYY")}
      </p>
    </div>
    <div className="uk-card-media-bottom">
      <img
        className="article-list__image"
        uk-img={`dataSrc: ${props.post.coverPhotoURL}`}
      />
    </div>
  </div>
);

export default ArticleListItem;
