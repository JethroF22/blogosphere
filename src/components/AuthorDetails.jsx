import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faUserCircle } from "@fortawesome/free-solid-svg-icons";

const AuthorDetails = props => (
  <div className="details">
    {props.photo ? (
      <div className="details__left">
        <img src={props.photo} />
      </div>
    ) : (
      <FontAwesomeIcon icon={faUserCircle} size="4x" />
    )}
    <div className="details__right">
      <p>Published by {props.username}</p>
      <button
        className="uk-button uk-button-default button"
        onClick={props.followed ? null : props.followAuthor}
        disabled={props.followed}
      >
        {props.followed ? `Following ${props.username}` : `Followed`}
      </button>
    </div>
  </div>
);

export default AuthorDetails;
