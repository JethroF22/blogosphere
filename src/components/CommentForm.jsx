import React, { Component } from "react";
import { connect } from "react-redux";

import { createComment } from "../actions/blog";

class CommentForm extends Component {
  state = {
    commentBody: ""
  };

  onCommentChange = e => {
    const commentBody = e.target.value;
    this.setState(() => ({ commentBody }));
  };

  onSubmit = e => {
    e.preventDefault();

    this.props
      .createComment(
        this.state.commentBody,
        this.props.postID,
        this.props.token
      )
      .then(slug => {
        if (this.props.actionStatus === "Action successful") {
          this.setState(() => ({
            commentBody: ""
          }));
        }
      });
  };

  render() {
    return (
      <div className="form">
        <form className="uk-form-stacked" onSubmit={this.onSubmit}>
          <label htmlFor="commentBody" className="form__label">
            Leave a comment
          </label>
          <input
            type="text"
            name="commentBody"
            value={this.state.commentBody}
            onChange={this.onCommentChange}
            placeholder="Let the author know what you think..."
            className={"uk-textarea form__textarea"}
          />
          <br />
          <button
            type="submit"
            className="uk-button uk-button-default button"
            disabled={this.state.commentBody ? false : true}
          >
            Submit comment
          </button>
        </form>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  token: state.auth.token,
  postID: state.blog.currentPost._id,
  actionStatus: state.status.status
});

const mapDispatchToProps = dispatch => ({
  createComment: (commentBody, postID, token) =>
    dispatch(createComment(commentBody, postID, token))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CommentForm);
