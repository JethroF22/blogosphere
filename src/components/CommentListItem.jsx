import React from "react";
import moment from "moment";

const CommentListItem = props => (
  <div>
    <h3>
      {props.comment.username} ({moment(props.comment.timestamp).format(
        "Do MMM"
      )})
    </h3>
    <p>{props.comment.body}</p>
  </div>
);

export default CommentListItem;
