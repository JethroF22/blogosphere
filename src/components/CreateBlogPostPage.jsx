import React, { Component } from "react";
import { connect } from "react-redux";
import Textarea from "react-textarea-autosize";

class CreateBlogPostPage extends Component {
  state = {
    body: "",
    title: "",
    coverPhotoURL: ""
  };

  render() {
    return (
      <div>
        <h1>Create a Blog Post</h1>
        <form>
          <label htmlFor="title">Title: </label>
          <input
            type="text"
            name="title"
            value={this.state.title}
            onChange={this.onTitleChange}
          />
          <br />
          <label htmlFor="body">Body: </label>
          <Textarea
            name="body"
            onChange={this.onBodyChange}
            value={this.state.body}
          />
          <br />
          <label htmlFor="coverPhoto">Cover Photo: </label>
          <input
            type="text"
            name="coverPhoto"
            value={this.state.coverPhotoURL}
            onChange={this.onUsernameChange}
            placeholder="Image URL"
          />
          <br />
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  createArticle: (article, token) => dispatch(createArticle(article, token))
});

export default connect(
  undefined,
  mapDispatchToProps
)(CreateBlogPostPage);
