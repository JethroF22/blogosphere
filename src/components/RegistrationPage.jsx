import React, { Component } from "react";

class RegistrationPage extends Component {
  state = {
    password: "",
    confirmPassword: "",
    username: "",
    email: ""
  };

  render() {
    return (
      <div>
        <h1>Create An Account</h1>
        <form>
          <label htmlFor="username">Username: </label>
          <input type="text" name="username" />
          <br />
          <label htmlFor="email">Email: </label>
          <input type="text" name="email" />
          <br />
          <label htmlFor="password">Password: </label>
          <input type="password" name="password" />
          <br />
          <label htmlFor="confirmPassword">Confirm Password: </label>
          <input type="password" name="confirmPassword" />
          <br />
        </form>
      </div>
    );
  }
}

export default RegistrationPage;
