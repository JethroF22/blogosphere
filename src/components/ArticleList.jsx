import React from "react";
import { Link } from "react-router-dom";

import ArticleListItem from "./ArticleListItem";

const ArticleList = props => (
  <div className="article-list">
    {props.posts.map(post => (
      <Link
        to={`/blog/view/${post.slug}`}
        key={Math.floor(Math.random() * 99999)}
      >
        <ArticleListItem post={post} />
      </Link>
    ))}
  </div>
);

export default ArticleList;
