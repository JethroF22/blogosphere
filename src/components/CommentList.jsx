import React from "react";
import CommentListItem from "./CommentListItem";

const CommentList = props => (
  <div className="comments">
    {props.comments.map(comment => <CommentListItem comment={comment} />)}
  </div>
);

export default CommentList;
