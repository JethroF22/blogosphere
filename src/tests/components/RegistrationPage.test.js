import React from "react";
import { shallow } from "enzyme";
import _ from "lodash";

import { RegistrationPage } from "../../components/RegistrationPage";
import users from "../seed/users";

let wrapper, register;

beforeEach(() => {
  register = jest.fn();
  wrapper = shallow(<RegistrationPage register={register} />);
});

test("Should render component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("Should submit form with valid data", () => {
  wrapper.setState({ ...users[0] });
  wrapper.find("form").simulate("submit", { preventDefault: () => {} });
  expect(register).toHaveBeenLastCalledWith(
    _.pick(users[0], ["email", "password", "username"])
  );
});

test("Should not submit form with invalid data", () => {
  wrapper.setState({ ...users[1] });
  wrapper.find("form").simulate("submit", { preventDefault: () => {} });
  expect(wrapper.state("errors")).toMatchObject({
    confirmPassword: "Passwords do not match",
    email: "Not a valid email"
  });
});
