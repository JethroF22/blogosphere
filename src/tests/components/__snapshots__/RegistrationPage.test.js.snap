// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component correctly 1`] = `
<div>
  <h1>
    Create An Account
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="username"
    >
      Username: 
    </label>
    <input
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <label
      htmlFor="email"
    >
      Email: 
    </label>
    <input
      name="email"
      onChange={[Function]}
      type="text"
      value=""
    />
    <br />
    <label
      htmlFor="password"
    >
      Password: 
    </label>
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <br />
    <label
      htmlFor="confirmPassword"
    >
      Confirm Password: 
    </label>
    <input
      name="confirmPassword"
      onChange={[Function]}
      type="password"
      value=""
    />
    <br />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
